src/Adapter/Order/Refund/OrderProductRemover.php

```diff
@@ -0,0 +1,265 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Order\Refund;
+
+use Cart;
+use Configuration;
+use Context;
+use Db;
+use Order;
+use OrderDetail;
+use OrderHistory;
+use PrestaShop\PrestaShop\Core\Domain\Order\Exception\DeleteCustomizedProductFromOrderException;
+use PrestaShop\PrestaShop\Core\Domain\Order\Exception\DeleteProductFromOrderException;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class OrderProductRemover
+{
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * OrderProductRemover constructor.
+     *
+     * @param LoggerInterface $logger
+     */
+    public function __construct(LoggerInterface $logger, TranslatorInterface $translator)
+    {
+        $this->logger = $logger;
+        $this->translator = $translator;
+    }
+
+    /**
+     * @param Order $order
+     * @param OrderDetail $orderDetail
+     * @param int $quantity
+     * @param bool $isCustomized
+     */
+    public function deleteProductFromOrder(Order $order, OrderDetail $orderDetail, int $quantity, $isCustomized = false)
+    {
+        if ($isCustomized) {
```

Comment:
You don't need this boolean since you have the info in `$orderDetail->id_customization`